### PR TITLE
fix(payload): Ethereum Payload Builder Zero Type

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Defines an `EthereumPayloadBuilder` type as the empty tuple type following @gakonst's comment in https://github.com/paradigmxyz/reth/pull/4377#discussion_r1382498942

### CODE REVIEW

File: crates/payload/basic/src/lib.rs [hunk 1]
```diff
@@ -57,6 +57,9 @@ use tracing::{debug, trace};
 
 mod metrics;
 
+#[cfg(not(feature = "optimism"))]
+pub type EthereumPayloadBuilder = ();
```
Comment (mattsse) [inline crates/payload/basic/src/lib.rs:61]: I'd like to not use a type alias here and instead a zero sized new struct for this

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/payload/basic/src/lib.rs [hunk 2]
```diff
@@ -57,6 +57,10 @@ use tracing::{debug, trace};
 
 mod metrics;
 
+/// Ethereum payload builder
+#[cfg(not(feature = "optimism"))]
```
Comment (mattsse) [inline crates/payload/basic/src/lib.rs:61]:
```suggestion
#[derive(Debug,Clone, Default)
#[non_exhaustive]
```

**Review by mattsse — COMMENTED** [review]
the default usage also needs to be updated:

https://github.com/paradigmxyz/reth/blob/3d076b1594139f809a1483a9027bafbe5bf1105d/crates/payload/basic/src/lib.rs#L67-L67

File: crates/payload/basic/src/lib.rs (hunk 2 above)
Comment (refcell) [reply crates/payload/basic/src/lib.rs]: Added without `non_exhaustive` since it cannot be constructed outside of the defining crate with the attribute

**Review by mattsse — APPROVED** [review]

**Pull request merged**